feat: support clone by reference (#13)
fastify/deepmerge: README.md

console.log(resultDeepmergedArray) // [ { a: [ 1, 2, 3 ], b: [ 4, 5, 6 ] } ]
```

#### cloneProtoObject

> Merging objects with prototypes such Streams or Buffer, it is not supported by default.

Reviewer: ```suggestion
Merging objects with prototypes, such as Streams or Buffers, are not supported by default.
```
